common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/ElasticRetriever.scala

```diff
@@ -20,7 +20,13 @@ class ElasticRetriever[T](client: ElasticClient, index: Index)(
     extends Retriever[T]
     with Logging {
 
-  override final def apply(ids: Seq[String]): Future[RetrieverMultiResult[T]] =
+  override final def apply(
+    ids: Seq[String]): Future[RetrieverMultiResult[T]] = {
```

Comment:
Just as a thought experiments, as I know that it would have larger knock-on effects, but could this be a [`NonEmptyList`](https://typelevel.org/cats/datatypes/nel.html). We use them [in the merger for lists of work](https://github.com/wellcomecollection/catalogue/blob/48be8c35d1e0299c1ab43e225fa6b164b328c9df/pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/rules/FieldMergeRule.scala#L48) and would encapsulate this requirement in the type system.

Reply:
Don't let stray `cats` in.

Reply:
I would say `NonEmptyList` makes sense. We already have cats every classpath

Reply:
Maybe if I was embarking on a more full-blown refactor of the matcher, but I'm not sure I want to commit to that work right now?

Reply:
yeah. to be honest , on second thought, you still have to create a `NonEmptyList` from a `List` which means checking the `List` elsewhere and having that duplicated code in all usages of `ElasticRetriever` which is probably annoying

Reply:
> Maybe if I was embarking on a more full-blown refactor of the matcher, but I'm not sure I want to commit to that work right now?

Indeedy, hence "thought experiment" - glad we dabbled.